### ISSUE TITLE
wpt run crashes on tests with firefox

### ISSUE THREAD

**Issue by ziransun** [web-platform-tests/wpt#24991]
Codebase: master commit header e9e85e67994df01673a8eb55c47d1b6f5eb6c18a
Command: `./wpt  run --yes --no-manifest-update --manifest MANIFEST.json  --log-mach-level=debug --log-mach=- firefox dom/historical.html`

Logs:
 ```
0:00.03 INFO Using certutil /usr/bin/certutil
 0:00.03 INFO Using webdriver binary /home/ziran/WPT/wpt/_venv2/bin/geckodriver
 0:00.06 WARNING Supplied channel doesn't match binary, using supplied channel
 0:00.06 INFO Using cached test prefs from /home/ziran/WPT/wpt/_venv2/profiles/nightly/79.0
 0:00.06 INFO Running in headless mode, pass --no-headless to disable
 0:00.17 mozversion INFO application_buildid: 20200720193547
 0:00.17 mozversion INFO application_changeset: df3ed76cf46b23c9b658cd5be4cdd4162d86f736
 0:00.17 mozversion INFO application_display_name: Firefox
 0:00.17 mozversion INFO application_id: {ec8030f7-c20a-464f-9b0e-13a3a9e97384}
 0:00.17 mozversion INFO application_name: Firefox
 0:00.17 mozversion INFO application_remotingname: firefox
 0:00.17 mozversion INFO application_repository: https://hg.mozilla.org/releases/mozilla-release
 0:00.17 mozversion INFO application_vendor: Mozilla
 0:00.17 mozversion INFO application_version: 79.0
 0:00.17 mozversion INFO platform_buildid: 20200720193547
 0:00.17 mozversion INFO platform_changeset: df3ed76cf46b23c9b658cd5be4cdd4162d86f736
 0:00.17 mozversion INFO platform_repository: https://hg.mozilla.org/releases/mozilla-release
 0:00.17 mozversion INFO platform_version: 79.0
 0:00.18 vcs DEBUG git rev-parse --show-toplevel
 0:00.18 vcs DEBUG git rev-parse --show-cdup
 0:00.18 vcs DEBUG git rev-parse HEAD
 0:01.02 INFO Using 1 client processes
 0:01.06 INFO Starting http server on 127.0.0.1:8000
 0:01.06 INFO Starting http2 server on 127.0.0.1:9000
 0:01.06 INFO Starting http server on 127.0.0.1:8001
 0:01.07 INFO Starting https server on 127.0.0.1:8443
 0:01.07 INFO Starting https server on 127.0.0.1:8444
 0:01.56 SUITE_START: web-platform-test - running 1 tests
 0:01.56 INFO Running reftest tests
 0:01.57 DEBUG Using 1 processes
 0:01.57 INFO No reftest tests to run
 0:01.57 DEBUG Stop flag set in ManagerGroup
 0:01.57 INFO Running print-reftest tests
 0:01.58 DEBUG Using 1 processes
 0:01.58 INFO No print-reftest tests to run
 0:01.58 DEBUG Stop flag set in ManagerGroup
 0:01.58 INFO Running wdspec tests
 0:01.58 DEBUG Using 1 processes
 0:01.58 INFO No wdspec tests to run
 0:01.58 DEBUG Stop flag set in ManagerGroup
 0:01.58 INFO Running testharness tests
 0:01.58 DEBUG Using 1 processes
 0:01.63 INFO Setting up ssl
 0:01.64 certutil Full command: /usr/bin/certutil -N -d /tmp/tmpoODsZl.mozrunner -f /tmp/tmpoODsZl.mozrunner/.crtdbpw
certutil 
 0:01.65 certutil 
 0:01.66 certutil 
Certificate Nickname                                         Trust Attributes
                                                             SSL,S/MIME,JAR/XPI

web-platform-tests                                           CT,, 

 0:01.66 DEBUG Dispatch start_init
 0:01.66 DEBUG new state: initializing_browser
 0:01.66 DEBUG Dispatch init
 0:01.66 DEBUG Init called, starting browser and runner
 0:01.66 DEBUG Starting browser with settings {'lsan_allowed': set([]), 'mozleak_thresholds': False, 'lsan_max_stack_depth': None, 'check_leaks': False, 'lsan_disabled': False, 'mozleak_allowed': False}
 0:01.66 DEBUG Starting Firefox
 0:01.66 INFO Application command: /usr/bin/firefox --marionette about:blank -profile /tmp/tmpvuYDvD
 0:01.66 DEBUG Firefox Started
 0:01.67 DEBUG Starting Firefox
 0:01.67 INFO Application command: /usr/bin/firefox --marionette about:blank -profile /tmp/tmpKPlo2c
 0:01.67 DEBUG Firefox Started
 0:01.67 INFO Starting runner
 0:01.67 DEBUG Test runner started
 0:01.68 DEBUG Got command: u'log'
 0:01.68 DEBUG Executor setup
 0:01.68 DEBUG Got command: u'log'
 0:01.68 DEBUG Connecting to Marionette on port 47033
 0:01.68 DEBUG Got command: u'log'
 0:01.68 DEBUG Waiting for Marionette connection
 0:01.69 pid:14156 Full command: /usr/bin/firefox --marionette about:blank -profile /tmp/tmpvuYDvD
pid:14156 *** You are running in headless mode.
 0:02.81 pid:14156 console.error: SearchCache: "_readCacheFile: Error reading cache file:" (new Error("", "(unknown module)"))
 0:03.31 pid:14156 1597313046103	Marionette	INFO	Listening on port 47033
 0:03.39 DEBUG Got command: u'log'
 0:03.39 DEBUG Starting Marionette session
 0:03.42 DEBUG Got command: u'log'
 0:03.42 DEBUG Marionette session started
 0:03.42 DEBUG Got command: u'log'
 0:03.42 DEBUG Loading http://web-platform.test:8000/testharness_runner.html
 0:03.50 DEBUG Got command: u'init_succeeded'
 0:03.50 DEBUG new state: running
 0:03.50 DEBUG Dispatch run_test
 0:03.50 TEST_START: /dom/historical.html
 0:03.50 DEBUG Got command: u'log'
 0:03.50 DEBUG Executor setup done
 0:03.51 DEBUG Got command: u'log'
 0:03.51 INFO NoSuchWindowException on command, setting status to CRASH
 0:03.51 DEBUG Got command: 'test_ended'
 0:03.51 DEBUG Unexpected count in this thread 1
 0:03.51 TEST_END: CRASH, expected OK
 0:03.51 INFO No more tests
 0:03.51 DEBUG new state: stop
 0:03.51 DEBUG TestRunnerManager main loop terminating, starting cleanup
 0:03.51 DEBUG Stopping Firefox 14156
 0:03.54 pid:14156 1597313046331	Marionette	INFO	Stopped listening on port 47033
 0:03.82 INFO Browser exited with return code 0
 0:03.82 INFO PROCESS LEAKS None
 0:03.82 DEBUG stopped
 0:03.82 DEBUG ensure_runner_stopped
 0:03.82 DEBUG stopped
 0:03.82 DEBUG waiting for runner process to end
 0:03.82 DEBUG After join
 0:03.82 DEBUG Runner process exited with code 0
 0:03.82 DEBUG TestRunnerManager cleanup
 0:03.82 INFO Closing logging queue
 0:03.82 INFO queue closed
 0:03.82 DEBUG TestRunnerManager teardown
 0:03.82 INFO PROCESS LEAKS None
 0:03.83 DEBUG Stopping Firefox 14162
 0:03.84 INFO Browser exited with return code -15
 0:03.85 DEBUG TestRunnerManager main loop terminated
 0:03.85 DEBUG Stop flag set in ManagerGroup
 0:03.85 INFO Running crashtest tests
 0:03.85 DEBUG Using 1 processes
 0:03.85 INFO No crashtest tests to run
 0:03.85 DEBUG Stop flag set in ManagerGroup
 0:03.85 INFO Got 1 unexpected results
 0:03.85 SUITE_END

web-platform-test
~~~~~~~~~~~~~~~~~
Ran 1 checks (1 tests)
Expected results: 0
Unexpected results: 1
  test: 1 (1 crash)

Unexpected Results
------------------
CRASH /dom/historical.html
 0:03.88 INFO Closing logging queue
 0:03.88 INFO queue closed
```

**Comment by stephenmcgruer** [comment]
Hrm, I cannot reproduce *but* I notice I am using a locally installed version of Firefox:

```
./wpt run --yes --no-manifest-update --manifest MANIFEST.json --log-mach-level=debug --log-mach=- firefox dom/historical.html
...
 0:02.47 INFO Application command: /home/smcgruer/github/wpt/_venv2/browsers/nightly/firefox/firefox --marionette about:blank -profile /tmp/tmpNaXt60
```

(Versus your /usr/bin/firefox).

One thing that's interesting here is the log line: `0:00.06 WARNING Supplied channel doesn't match binary, using supplied channel`. Notably, Firefox 79 is the stable release, but the default for `channel` nowadays is nightly (see `0:00.06 INFO Using cached test prefs from /home/ziran/WPT/wpt/_venv2/profiles/nightly/79.0`). I wonder if there's a conflict here.

**Comment by stephenmcgruer** [comment]
Ahah, yes, I can reproduce by forcing a stable binary for Firefox but letting the channel auto-select nightly:

```
./wpt run --binary=~/firefox-stable/firefox --no-manifest-update --manifest MANIFEST.json --log-mach-level=debug --log-mach=- firefox dom/historical.html
...
 0:09.42 INFO NoSuchWindowException on command, setting status to CRASH
 0:09.42 DEBUG Got command: 'test_ended'
 0:09.43 DEBUG Unexpected count in this thread 1
 0:09.43 TEST_END: CRASH, expected OK
```

**Comment by stephenmcgruer** [comment]
I suspect this is likely a duplicate of https://github.com/web-platform-tests/wpt/issues/24924, but let me be sure before we close anything.

**Comment by stephenmcgruer** [comment]
Hrm ok, it's not a mismatch problem given that even adding `--channel=stable` doesn't fix it.

**Comment by Hexcles** [comment]
Oh, thanks for pinning down a concrete example that triggers the crash consistently!

**Comment by stephenmcgruer** [comment]
@ziransun I believe I just fixed this in https://github.com/web-platform-tests/wpt/issues/24924 - can you try sync-ing to HEAD and re-running, to see if you still get a crash?

**Comment by ziransun** [comment]
@stephenmcgruer: it runs fine now. Thank you!

**Comment by ziransun** [comment]
Close the issue as it's fixed in  #24924
